src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+		return (SubmissionSampleLocation[]) list.toArray(new SubmissionSampleLocation[list.size()]);
+	}
+	
+	/**
+	 * Update selected languages label and selected language list
+	 * @param valuesList
+	 */
+	protected void updateLanguagesLabel(List<Object> valuesList) {
+	    if (valuesList == null || valuesList.size() == 0) {
+	        selectedLanguagesLabel.setText("None selected");
+	        selectedLanguagesLabel.setToolTipText("None selected");
+        } else {
+            selectedLanguagesLabel.setText(valuesList.size()+" selected");
+            selectedLanguagesLabel.setToolTipText(Arrays.toString(valuesList.toArray()));
+            
+            selectedLanguageList = new ArrayList<Language>();
+            valuesList.forEach((lang) -> {selectedLanguageList.add((Language)lang);});
+        }
+	}
+
+	protected void updateProblemsLabel(List<Object> valuesList) {
+	    
+	    // TODO write code
+	    if (valuesList == null || valuesList.size() == 0) {
+            selectedProblemsLabel.setText("None selected");
+            selectedProblemsLabel.setToolTipText("None selected");
+        } else {
+            
+            selectedProblemsLabel.setText(valuesList.size()+" selected");
+            selectedProblemsLabel.setToolTipText(Arrays.toString(valuesList.toArray()));
+            
+            selectedProblemList = new ArrayList<Problem>();
+            valuesList.forEach((prob) -> {selectedProblemList.add((Problem)prob);});
+        }
+	}
+
+	protected void updateNosLabel(List<Object> valuesList) {
+
+        if (valuesList == null || valuesList.size() == 0) {
+            selectedNosLabel.setText("None selected");
+            selectedNosLabel.setToolTipText("None selected");
+        } else {
+
+            selectedNosLabel.setText(valuesList.size() + " selected");
+            selectedNosLabel.setToolTipText(Arrays.toString(valuesList.toArray()));
+            submissionSolutionList = new SubmissionSolutionList();
+
+            // update/load selected items into submissionSolutionList
+            valuesList.forEach((subSL) -> {
+                submissionSolutionList.add((SubmissionSampleLocation) subSL);
+            });
+
+        }
+    }
+
+	protected void clearTextArea() {
+		textArea.selectAll();
+		textArea.replaceSelection("");
+		linenumber = 0;
+	}
+
+	void addLineToTextArea(String s) {
+
+		linenumber++;
+		String fmtLineNumber = String.format("%4d", linenumber);
+
+		textArea.insert(fmtLineNumber + " " + hhMMSSformatter.format(new Date()) + " " + s + "\n", 0);
+//		textArea.append(s);
+	}
+
+	protected void selectNewCDP() {
+
+		File file = selectEntry("Select CDP");
+
+		File cdpDir = FileUtilities.findCDPConfigDirectory(file);
+
+		if (cdpDir != null) {
+		    updateCDPDirAndFields(cdpDir.getAbsolutePath());
+		} else {
+
+			int result = FrameUtilities.yesNoCancelDialog(this,
+					file.getAbsoluteFile() + " may not be a CDP directory, continue anyways?", "Select CDP");
```

Comment:
If the call to `selectEntry()` (line 590, above) returns null (which it can do if the `JFileChooser` return option inside `selectEntry()`  is `CANCEL` or `ERROR`), and if the call to `findCDPConfigDirectory()` (line 592, above) also returns null (which it can do if the specified "location" is neither a directory nor a YAML file), then it appears to me that the reference on the above line to `file.getAbsoluteFile()` will throw a `NullPointerException`.  I think that needs to be handled (perhaps via a try/catch?)